### ISSUE TITLE
Separate Admin Groups for each Workload 

### ISSUE THREAD

**Issue by TheKoguryo** [oracle-quickstart/oci-landing-zones#85]
When I apply templates/enterprise-landing-zone, 3 user groups are created for first workload with related IAM policies.

- OCI-ELZ-UGP-N-APP-ADMIN
- OCI-ELZ-UGP-N-DB-ADMIN
- OCI-ELZ-UGP-N-WRK-ADMIN

And then when I apply templates/elz-workload for another workload, exactly same 3 user groups are created.

- OCI-ELZ-UGP-N-APP-ADMIN
- OCI-ELZ-UGP-N-DB-ADMIN
- OCI-ELZ-UGP-N-WRK-ADMIN

For isolate each workload, it is required that separate Workload Admin groups manage only resources under their own Workload compartment.

**Comment by JSAhlawat** [comment]
Created and will track this issue under [LANZ-1971]([LANZ-1971](https://jira.oci.oraclecorp.com/browse/LANZ-1971))

**Comment by VinayKumar611** [comment]
Hi @TheKoguryo, 

Please find the fix of the issue on the file iam.tf from this branch  https://github.com/oracle-quickstart/oci-landing-zones/blob/integration/templates/elz-workload/iam.tf. 

Unique Workload Group name will be achieved by using the variable "workload_prefix" on the tfvars file across multiple workload deployment . So , make sure to change this varibale after every workload deployment. 

PS :: This is Alpha Release Branch , so no testing has been done . Will be prone to some unwanted bugs. 

Please let us know your feedback on the fix.

**Comment by VinayKumar611** [comment]
Hi @TheKoguryo , 

Please enter the valid IDCS endpoint. 

Proper Syntax for IDCS :: ^https://idcs-.*.identity.oraclecloud.com:443$

Use this documentation to get the identity domain idcs endpoint. 
https://docs.oracle.com/en-us/iaas/Content/Identity/domains/to-view-details-of-an-identity-domain.htm?Highlight=get%20domain%20idcs

**Comment by TheKoguryo** [comment]
Hi, @VinayKumar611,

I tested integration branch in Resource Manager.

For templates/enterprise-landing-zone, I add **idcs_endpoint** in templates/elz-environment/workload.tf
```
 26   identity_domain_name         = module.identity.domain.display_name
 27   idcs_endpoint                = module.identity.domain.url
 28   workload_admin_group_name    = var.workload_admin_group_name
```

For templates/elz-workload, I add **idcs_endpoint** in templates/elz-workload/schema.yaml
```
 56   - title: IAM Policy Variables
 57     visible: true
 58     variables:
 59       - identity_domain_id
 60       - identity_domain_name
 61       - idcs_endpoint
 62       - workload_admin_group_name
...
276   identity_domain_name:
277     type: string
278     description: "the name of identity domain that the workload expansion is going to create in"
279     title: Identity Domain Name
280     required: true
281   idcs_endpoint:
282     type: string
283     description: "the url of identity domain that the workload expansion is going to create in"
284     title: Identity Domain URL
285     required: true
```

**Comment by TheKoguryo** [comment]
Hi, @VinayKumar611,

Additionally, I found typo in integration branch.

Filename: /templates/elz-workload/iam.tf

Current:
  base_group_names = {
    network_admin_group_name : var.network_admin_group_name != "" ? var.network_admin_group_name : "**OOCI**-ELZ-UGP-${var.environment_prefix}-${var.workload_prefix}-NET-ADMIN",
  }

After:
  base_group_names = {
    network_admin_group_name : var.network_admin_group_name != "" ? var.network_admin_group_name : "**OCI**-ELZ-UGP-${var.environment_prefix}-${var.workload_prefix}-NET-ADMIN",
  }

**Comment by VinayKumar611** [comment]
Hi @TheKoguryo , 

Thanks for providing your feedback. We will fix these before we merge them on Release branch.

**Comment by VinayKumar611** [comment]
Fixed the issue on latest release v2.2.0.
